### PR TITLE
ugps: add quality measurement parameters

### LINKED ISSUE #2: ugps miss GPS quality parameters
ugps does not provide information about GPS signal quality. This is very impractical because the user cannot know what accuracy he can count on. The basic quality parameters are the number of visible satellites and the HDOP (Dilution of precision). Both can be found in the $GPGGA report, which is already being processed.

This PR https://github.com/openwrt/ugps/pull/1 adds support for both of these. Please include it in all affected development branches.

### PR DESCRIPTION
**Pull request by zokl**
The current version of ugps does not report any quality parameters for GPS receiver fixation. Considering the positioning error, which can be up to hundreds of meters, it is helpful to know how accurately the GPS determines the position. This PR supports determining the number of satellites and the HDOP (Horizontal Dilution of Precision) parameter to address this. The GPGGA sentence is already parsed in the code, so it just extends the amount of data it reads.

Fixes: #2 (ugps miss GPS quality parameters)

### CODE REVIEW

**Comment by zokl** [comment]
Any idea how to push it do master?

File: main.c
Comment (ynezz) [inline main.c]: This seems like unrelated change, so either add it as a separate fix commit or remove it for now. That change makes sense, because `-S` doesn't seems to be taking any arguments.

**Review by ynezz — CHANGES_REQUESTED** [review]
Just that `getopt()` change seems unrelated and needs to be reworked/removed, otherwise LGTM and I would merge it unless someone other beats me to it.

File: main.c
Comment (zokl) [reply main.c]: Thank you, Fixed.

**Comment by zokl** [comment]
Is it possible to push it to 22 and 21, too?

File: main.c
Comment (zokl) [reply main.c]: I made the change. Is it possible to accept it? @ynezz

**Comment by ynezz** [comment]
I just wanted to push that, but noticed:

```
From: John Crispin <john@phrozen.org>
...
Fixes: #2 (ugps miss GPS quality parameters)
Signed-off-by: Zbynek Kocur <zbynek.kocur@fel.cvut.cz>
```

Which basically means, that John is author of the changes, is that correct? If that is the case, then you need to adjust the footer and add John's `Signed-off-by` as well. 

```
Fixes: #2 (ugps miss GPS quality parameters)
Signed-off-by: John Crispin <john@phrozen.org>
Signed-off-by: Zbynek Kocur <zbynek.kocur@fel.cvut.cz>
```

If that is not case (just guessing from #1), then you should correct the authorship/From: by `git commit --amend --reset-author` (or such) to yourself.

**Comment by ynezz** [comment]
> Is it possible to push it to 22 and 21, too?

Well, 21.02 is EOL, 22.03 is almost EOL, so I doubt that. There is 23.05, but generally speaking the idea of stable maintenance branches is well a stability, so ideally just a fixes should be backported. Since this doesn't look like a fix, I wouldn't consider it for a backport.

**Comment by zokl** [comment]
> I just wanted to push that, but noticed:
> 
> ```
> From: John Crispin <john@phrozen.org>
> ...
> Fixes: #2 (ugps miss GPS quality parameters)
> Signed-off-by: Zbynek Kocur <zbynek.kocur@fel.cvut.cz>
> ```
> 
> Which basically means, that John is author of the changes, is that correct? If that is the case, then you need to adjust the footer and add John's `Signed-off-by` as well.
> 
> ```
> Fixes: #2 (ugps miss GPS quality parameters)
> Signed-off-by: John Crispin <john@phrozen.org>
> Signed-off-by: Zbynek Kocur <zbynek.kocur@fel.cvut.cz>
> ```
> 
> If that is not case (just guessing from #1), then you should correct the authorship/From: by `git commit --amend --reset-author` (or such) to yourself.

Thank you, I made change.

**Comment by zokl** [comment]
> > Is it possible to push it to 22 and 21, too?
> 
> Well, 21.02 is EOL, 22.03 is almost EOL, so I doubt that. There is 23.05, but generally speaking the idea of stable maintenance branches is well a stability, so ideally just a fixes should be backported. Since this doesn't look like a fix, I wouldn't consider it for a backport.

I understand that intention. My patch improves but is compatible with all previous versions. I would like to have it in the lower, still-used versions, but I understand.

**Comment by ynezz** [comment]
> but is compatible with all previous versions

Its not about backward compatibility, but more about a possible regressions, every change has certain potential for regressions, so the goal here is to limit the number of changes in the stable branches and basically just try to fix bugs.

**Comment by ynezz** [comment]
BTW there are some whitespace issues:

```
Applying: ugps: add quality measurement parameters
.git/rebase-apply/patch:40: space before tab in indent.
 	strncpy(hdop, nmea_params[8].str, sizeof(hdop));
.git/rebase-apply/patch:45: space before tab in indent.
 	DEBUG(4, "HDOP: %s\n", hdop);
warning: 2 lines add whitespace errors.
```

I've fixed them now myself, but would be nice to check for those next time, thanks. Merged in 69561a074d

**Comment by zokl** [comment]
> BTW there are some whitespace issues:
> 
> ```
> Applying: ugps: add quality measurement parameters
> .git/rebase-apply/patch:40: space before tab in indent.
>  	strncpy(hdop, nmea_params[8].str, sizeof(hdop));
> .git/rebase-apply/patch:45: space before tab in indent.
>  	DEBUG(4, "HDOP: %s\n", hdop);
> warning: 2 lines add whitespace errors.
> ```
> 
> I've fixed them now myself, but would be nice to check for those next time, thanks. Merged in [69561a0](https://github.com/openwrt/ugps/commit/69561a074d6f50f63b82608b19041e5eb2c605a9)

 I am so sorry, I'll be more careful next time.

**Pull request closed without merging**
